lib/service/customer_support_service.dart

```diff
@@ -498,4 +525,124 @@ class CustomerSupportServiceImpl extends CustomerSupportService {
       await createRenderingIssueReport(token, ["IPFS Loading"]);
     }
   }
+
+  @override
+  Future<List<Object>> getIssuesAndAnnouncement() async {
+    List<Object> result = [];
+    List<Issue> issues = await getIssues();
+    List<AnnouncementLocal> announcements =
+        await _announcementDao.getAnnouncements();
+
+    if (issues.isNotEmpty && announcements.isNotEmpty) {
+      for (var issue in issues) {
+        final announcement = announcements.firstWhereOrNull(
+            (element) => element.announcementContextId == issue.announcementID);
+        if (announcement != null) {
+          issue.announcement = announcement;
+          announcements.remove(announcement);
+        }
+      }
+    }
+    result.addAll(issues);
+    result.addAll(announcements);
+    numberOfIssuesInfo.value = [
+      result.length,
+      result
+          .where((element) => (element is Issue)
+              ? element.unread > 0
+              : (element is AnnouncementLocal)
+                  ? element.unread
+                  : false)
+          .length,
+    ];
+    result.sort((a, b) {
+      final aTimestamp = _getTimestamp(a);
+      final bTimestamp = _getTimestamp(b);
+      return bTimestamp.compareTo(aTimestamp);
+    });
+    return result;
+  }
+
+  int _getTimestamp(Object a) {
+    if (a is Issue) {
+      final timestamp =
+          a.draft?.createdAt ?? a.lastMessage?.timestamp ?? a.timestamp;
+      return timestamp.millisecondsSinceEpoch;
+    } else if (a is AnnouncementLocal) {
+      return a.createdAt;
+    } else {
+      return 0;
+    }
+  }
+
+  @override
+  Future<void> fetchAnnouncement() async {
+    final lastPullTime = _configurationService.getAnnouncementLastPullTime();
+    final announcements = await _announcementApi.getAnnouncements(
+        lastPullTime: lastPullTime ?? 0);
+    final metricClient = injector.get<MetricClientService>();
+    metricClient.addEvent(
```

Comment:
I see you call `fetchAnnouncement` in many places, this event might push many times with the same value.